plugins/aws-appmesh/plugin/commands.go

```diff
@@ -173,11 +173,12 @@ func (plugin *Plugin) setupEgressRules(
 	}
 
 	if config.EgressIgnoredPorts != "" {
-		err = iptable.Append("nat", egressChain, "-p", "tcp", "-m", "multiport", "--dports",
-			config.EgressIgnoredPorts, "-j", "RETURN")
-		if err != nil {
-			log.Errorf("Append rule for egressIgnoredPorts failed: %v", err)
-			return err
+		for _, port := range config.EgressIgnoredPorts {
+			err = iptable.Append("nat", egressChain, "-p", "tcp", "-m", "multiport", "--dports", port, "-j", "RETURN")
+			if err != nil {
```

Comment:
Before this change, if I had 15 ignored port, I'd have a single iptables rule and all traffic will hit that single rule. After this change, there will be 15 rules, possibly 15 times overhead per packet hitting this chain. Do we know what the perf impact is? Can we measure it?

Reply:
yeah unfortunately i dont have a good answer to the performance impact. what are your thoughts on creating a rule for every 15 ports, something like this:

```golang
chunk := []string{}

for i, port := range config.EgressIgnoredPorts {
    if (i+1)%15 == 0 {
        err = iptable.Append("nat", egressChain, "-p", "tcp", "-m", "multiport", "--dports", strings.Join(chunk, splitter), "-j", "RETURN")
        if err != nil {
            log.Errorf("Append rule for egressIgnoredPorts failed: %v", err)
            return err
        }

        chunk = []string{}
    }

    chunk = append(chunk, port)
}

if len(chunk) > 0 {
    err = iptable.Append("nat", egressChain, "-p", "tcp", "-m", "multiport", "--dports", strings.Join(chunk, splitter), "-j", "RETURN")
    if err != nil {
        log.Errorf("Append rule for egressIgnoredPorts failed: %v", err)
        return err
    }
}
```

Reply:
This looks like a good approach to me, even ignoring performance I think the resulting config will be nicer

Reply:
Thanks. This looks good.
Do we need to move the chunk=append(chunk,port) line to execute before the if check? Otherwise every 15th port will be dropped.